.github/workflows/backend_cd.yml

```
@@ -0,0 +1,80 @@
+name: Backend CD
+
+on:
+  push:
+    branches:
+      - develop
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      build-directory: ./backend
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: 17
+          distribution: temurin
+
+      - name: Gradle Caching
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Build BootJar
+        run: ./gradlew bootJar
+        working-directory: ${{ env.build-directory }}
+
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: code-zap-jar
+          path: backend/build/libs/*.jar
+
+  deploy:
+    needs: build
+    runs-on: self-hosted
+    steps:
+      - name: Get Artifact Download URL
+        id: get-artifact-url
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          artifacts=$(curl -H "Authorization: token $GITHUB_TOKEN" \
+                            -H "Accept: application/vnd.github.v3+json" \
+                            https://api.github.com/repos/${{ github.repository }}/actions/artifacts)
+          artifact_id=$(echo "$artifacts" | jq -r '.artifacts[0].id') # 가장 최근 아티팩트의 id를 가져옴
+          if [[ -z "$artifact_id" || "$artifact_id" == "null" ]]; then
+            echo "Artifact ID not found or invalid"
+            exit 1
+          fi
+          download_url="https://api.github.com/repos/${{ github.repository }}/actions/artifacts/$artifact_id/zip"
+          echo "DOWNLOAD_URL=${download_url}" >> $GITHUB_ENV
+
+      - name: Download Artifact
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          WORK_DIRECTORY: ${{ secrets.WORK_DIRECTORY }}
+        run: |
+          curl -L -o ${{ secrets.WORK_DIRECTORY }}/code-zap-jar.zip -H "Authorization: token $GITHUB_TOKEN" ${{ env.DOWNLOAD_URL }}
+
+      - name: Run Deploy Script
+        run: |
+          cd ${{ secrets.WORK_DIRECTORY }}
+          unzip -o code-zap-jar.zip
+          RUNNER_TRACKING_ID="" && ./deploy.sh
+
+      - name: Verify Deploy Succeed
+        run: |
+          sleep 10 # Ensure there is enough time for the application to start
+          pgrep -f 'java -jar' || { echo "Deploy Failed" }
```

```suggestion
      - name: Verify Deploy Succeed
        run: |
          sleep 3
          pgrep -f 'java -jar.code-zap.*\.jar' || echo "Deploy Failed"
```

1. 10초는 다소 긴 것 같아요. 3초로 줄여도 충분하겠습니다. 
2. 아직은 서버에서 직접 jar 파일을 실행하며 테스트할 일이 많을 것 같아 pgrep 명령어의 표현식을 개선하면 좋겠어요.
3. 쉘스크립트에서 중괄호는 여러 명령어를 하나로 묶기 위해 사용합니다. 여기서는 echo 명령어만 실행되고 있으므로 중괄호가 없어도 되겠네요~

참고) #97 추후 작업하겠습니다!